ECR login uses deprecated get-login command

## 🐛 Describe the bug
When attempting the `publish-all-aws` target, it fails saying that the `get-login` command is an invalid argument to `aws ecr`.

## 📜 To Reproduce
Steps to reproduce the behavior:
1. Clone `NASA-PDS/kdp` and `cd ./kdp/build`
2. Run `make publish-all-aws DOCKER_REPO=XXXXXXX.amazonaws.com AWS_CLI_REGION=XXXXX AWS_CLI_PROFILE=XXXX`
3. See the following:

```
make publish-all-aws DOCKER_REPO=XXXXXXX.amazonaws.com AWS_CLI_REGION=XXXXX AWS_CLI_PROFILE=XXXX
/Library/Developer/CommandLineTools/usr/bin/make -C ../operator  "DOCKER_REPO=\"XXXXXX.amazonaws.com\"" " AWS_CLI_PROFILE=\"XXXX\"" " AWS_CLI_REGION=\"XXXXX\"" publish-aws

usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
To see help text, you can run:

  aws help
  aws <command> help
  aws <command> <subcommand> help

aws: error: argument operation: Invalid choice, valid choices are:

batch-check-layer-availability           | batch-delete-image
batch-get-image                          | batch-get-repository-scanning-configuration
complete-layer-upload                    | create-pull-through-cache-rule
create-repository                        | delete-lifecycle-policy
delete-pull-through-cache-rule           | delete-registry-policy
delete-repository                        | delete-repository-policy
describe-image-replication-status        | describe-image-scan-findings
describe-images                          | describe-pull-through-cache-rules
describe-registry                        | describe-repositories
get-authorization-token                  | get-download-url-for-layer
get-lifecycle-policy                     | get-lifecycle-policy-preview
get-registry-policy                      | get-registry-scanning-configuration
get-repository-policy                    | initiate-layer-upload
list-images                              | list-tags-for-resource
put-image                                | put-image-scanning-configuration
put-image-tag-mutability                 | put-lifecycle-policy
put-registry-policy                      | put-registry-scanning-configuration
put-replication-configuration            | set-repository-policy
start-image-scan                         | start-lifecycle-policy-preview
tag-resource                             | untag-resource
upload-layer-part                        | get-login-password
wait                                     | help

make[1]: *** [aws-login] Error 252
make: *** [make-core] Error 2
```

## 🕵️ Expected behavior
<!-- A clear and concise description of what you expected to happen. -->
Should login to ECR successfully.

## 📚 Version of Software Used
<!-- Run `make version` on any KDP component to get this information. -->
```
$ make version
CORE:
kdp-operator: 0.1.0
kdp-datainput-base: 0.1.0
kdp-node-base: 0.1.0
EXAMPLES:
kdp-datainput-prefix-listing: 0.1.0
kdp-datainput-singleinput: 0.1.0
kdp-node-passthrough: 0.1.0
kdp-node-validate: 0.1.0
kdp-node-validate-directory: 0.1.0
```

## 🩺 Test Data / Additional context
<!-- If applicable, Add test data or any other context about the problem here .-->

## 🏞 Screenshots
<!-- If applicable, add screenshots to help explain your problem. -->

## 🖥 System Info
 - OS: macOS
 - Version 11.6.4

--- a/images/base/kdp-datainput-base/Makefile
+++ b/images/base/kdp-datainput-base/Makefile
@@ -77,14 +77,17 @@ endif
 # HELPERS
 
 # generate script to login to aws docker repo
-CMD_REPOLOGIN := "eval $$\( aws ecr"
+CMD_REPOLOGIN := "aws ecr get-login-password"
 ifdef AWS_CLI_PROFILE
 CMD_REPOLOGIN += " --profile $(AWS_CLI_PROFILE)"
 endif
 ifdef AWS_CLI_REGION
 CMD_REPOLOGIN += " --region $(AWS_CLI_REGION)"
 endif
-CMD_REPOLOGIN += " get-login --no-include-email \)"
+CMD_REPOLOGIN += " | docker login --username AWS --password-stdin"
+ifdef DOCKER_REPO
+CMD_REPOLOGIN += " $(DOCKER_REPO)"
+endif
 
 # login to AWS-ECR
 aws-login: ## Auto login to AWS-ECR using aws-cli

--- a/images/base/kdp-node-base/Makefile
+++ b/images/base/kdp-node-base/Makefile
@@ -79,14 +79,17 @@ endif
 # HELPERS
 
 # generate script to login to aws docker repo
-CMD_REPOLOGIN := "eval $$\( aws ecr"
+CMD_REPOLOGIN := "aws ecr get-login-password"
 ifdef AWS_CLI_PROFILE
 CMD_REPOLOGIN += " --profile $(AWS_CLI_PROFILE)"
 endif
 ifdef AWS_CLI_REGION
 CMD_REPOLOGIN += " --region $(AWS_CLI_REGION)"
 endif
-CMD_REPOLOGIN += " get-login --no-include-email \)"
+CMD_REPOLOGIN += " | docker login --username AWS --password-stdin"
+ifdef DOCKER_REPO
+CMD_REPOLOGIN += " $(DOCKER_REPO)"
+endif
 
 # login to AWS-ECR
 aws-login: ## Auto login to AWS-ECR using aws-cli

--- a/images/examples/datainputs/kdp-datainput-prefix-listing/Makefile
+++ b/images/examples/datainputs/kdp-datainput-prefix-listing/Makefile
@@ -77,14 +77,17 @@ endif
 # HELPERS
 
 # generate script to login to aws docker repo
-CMD_REPOLOGIN := "eval $$\( aws ecr"
+CMD_REPOLOGIN := "aws ecr get-login-password"
 ifdef AWS_CLI_PROFILE
 CMD_REPOLOGIN += " --profile $(AWS_CLI_PROFILE)"
 endif
 ifdef AWS_CLI_REGION
 CMD_REPOLOGIN += " --region $(AWS_CLI_REGION)"
 endif
-CMD_REPOLOGIN += " get-login --no-include-email \)"
+CMD_REPOLOGIN += " | docker login --username AWS --password-stdin"
+ifdef DOCKER_REPO
+CMD_REPOLOGIN += " $(DOCKER_REPO)"
+endif
 
 # login to AWS-ECR
 aws-login: ## Auto login to AWS-ECR using aws-cli

--- a/images/examples/datainputs/kdp-datainput-singleinput/Makefile
+++ b/images/examples/datainputs/kdp-datainput-singleinput/Makefile
@@ -77,14 +77,17 @@ endif
 # HELPERS
 
 # generate script to login to aws docker repo
-CMD_REPOLOGIN := "eval $$\( aws ecr"
+CMD_REPOLOGIN := "aws ecr get-login-password"
 ifdef AWS_CLI_PROFILE
 CMD_REPOLOGIN += " --profile $(AWS_CLI_PROFILE)"
 endif
 ifdef AWS_CLI_REGION
 CMD_REPOLOGIN += " --region $(AWS_CLI_REGION)"
 endif
-CMD_REPOLOGIN += " get-login --no-include-email \)"
+CMD_REPOLOGIN += " | docker login --username AWS --password-stdin"
+ifdef DOCKER_REPO
+CMD_REPOLOGIN += " $(DOCKER_REPO)"
+endif
 
 # login to AWS-ECR
 aws-login: ## Auto login to AWS-ECR using aws-cli

--- a/images/examples/nodes/kdp-node-passthrough/Makefile
+++ b/images/examples/nodes/kdp-node-passthrough/Makefile
@@ -77,14 +77,17 @@ endif
 # HELPERS
 
 # generate script to login to aws docker repo
-CMD_REPOLOGIN := "eval $$\( aws ecr"
+CMD_REPOLOGIN := "aws ecr get-login-password"
 ifdef AWS_CLI_PROFILE
 CMD_REPOLOGIN += " --profile $(AWS_CLI_PROFILE)"
 endif
 ifdef AWS_CLI_REGION
 CMD_REPOLOGIN += " --region $(AWS_CLI_REGION)"
 endif
-CMD_REPOLOGIN += " get-login --no-include-email \)"
+CMD_REPOLOGIN += " | docker login --username AWS --password-stdin"
+ifdef DOCKER_REPO
+CMD_REPOLOGIN += " $(DOCKER_REPO)"
+endif
 
 # login to AWS-ECR
 aws-login: ## Auto login to AWS-ECR using aws-cli

--- a/images/examples/nodes/kdp-node-validate-directory/Makefile
+++ b/images/examples/nodes/kdp-node-validate-directory/Makefile
@@ -77,14 +77,17 @@ endif
 # HELPERS
 
 # generate script to login to aws docker repo
-CMD_REPOLOGIN := "eval $$\( aws ecr"
+CMD_REPOLOGIN := "aws ecr get-login-password"
 ifdef AWS_CLI_PROFILE
 CMD_REPOLOGIN += " --profile $(AWS_CLI_PROFILE)"
 endif
 ifdef AWS_CLI_REGION
 CMD_REPOLOGIN += " --region $(AWS_CLI_REGION)"
 endif
-CMD_REPOLOGIN += " get-login --no-include-email \)"
+CMD_REPOLOGIN += " | docker login --username AWS --password-stdin"
+ifdef DOCKER_REPO
+CMD_REPOLOGIN += " $(DOCKER_REPO)"
+endif
 
 # login to AWS-ECR
 aws-login: ## Auto login to AWS-ECR using aws-cli

--- a/images/examples/nodes/kdp-node-validate/Makefile
+++ b/images/examples/nodes/kdp-node-validate/Makefile
@@ -77,14 +77,17 @@ endif
 # HELPERS
 
 # generate script to login to aws docker repo
-CMD_REPOLOGIN := "eval $$\( aws ecr"
+CMD_REPOLOGIN := "aws ecr get-login-password"
 ifdef AWS_CLI_PROFILE
 CMD_REPOLOGIN += " --profile $(AWS_CLI_PROFILE)"
 endif
 ifdef AWS_CLI_REGION
 CMD_REPOLOGIN += " --region $(AWS_CLI_REGION)"
 endif
-CMD_REPOLOGIN += " get-login --no-include-email \)"
+CMD_REPOLOGIN += " | docker login --username AWS --password-stdin"
+ifdef DOCKER_REPO
+CMD_REPOLOGIN += " $(DOCKER_REPO)"
+endif
 
 # login to AWS-ECR
 aws-login: ## Auto login to AWS-ECR using aws-cli

--- a/operator/Makefile
+++ b/operator/Makefile
@@ -78,14 +78,17 @@ endif
 # HELPERS
 
 # generate script to login to aws docker repo
-CMD_REPOLOGIN := "eval $$\( aws ecr"
+CMD_REPOLOGIN := "aws ecr get-login-password"
 ifdef AWS_CLI_PROFILE
 CMD_REPOLOGIN += " --profile $(AWS_CLI_PROFILE)"
 endif
 ifdef AWS_CLI_REGION
 CMD_REPOLOGIN += " --region $(AWS_CLI_REGION)"
 endif
-CMD_REPOLOGIN += " get-login --no-include-email \)"
+CMD_REPOLOGIN += " | docker login --username AWS --password-stdin"
+ifdef DOCKER_REPO
+CMD_REPOLOGIN += " $(DOCKER_REPO)"
+endif
 
 # login to AWS-ECR
 aws-login: ## Auto login to AWS-ECR using aws-cli

